### PR TITLE
Master exception handling

### PR DESCRIPTION
**Pull request by joeyfreeman**
Changed plottalkexceptions.h. Exceptions can now be thrown as:
_throw NotFound("Unable to find episode");_
Uncaught exceptions will now be caught by main and displayed as a message box to provide more info than the previous abort() unhandled exception message.
Changed getSeason() and getEpisode() to return pointers.
Changed selectedShow, selectedSeason, and selectedEpisode to use pointers so changes are reflected throughout the application. Previously these variables could easily fall out of sync with each other and with tvShowMap. I tried to use references to accomplish this but it wasn't working.

### CODE REVIEW

**Review by haleyminshell — APPROVED** [review]
Code looks good. Pulled down the branch and all unit tests passed and I was able to run the application and saw expected behavior.

File: PlotTalk/PlotTalk.pro [hunk 1]
```diff
@@ -80,7 +80,8 @@ test {
         Test/testaccountmanager.h \
         Test/testreply.h \
         Test/testuser.h \
-	Test/testseason.h
+		Test/testseason.h \
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro]: Indentation seems a bit off.

File: PlotTalk/Src/dashboard.cpp [hunk 2]
```diff
@@ -122,7 +122,7 @@ void Dashboard::on_AboutButton_clicked()
 void Dashboard::on_leftList_itemClicked(QListWidgetItem *item)
 {
     QString selected = item->text();
-    selectedShow = DatabaseManagerSingleton::Instance().getTvShow(selected);
+    selectedShow = &DatabaseManagerSingleton::Instance().getTvShow(selected);
```
Comment (allanhaywood) [inline PlotTalk/Src/dashboard.cpp]: Yes, this should be the right way to make sure you are operating with a reference, this is called an independent reference.

I'm still a bit fuzzy regarding proper handling of references to the internal datamembers of a container class.

File: PlotTalk/Src/season.cpp [hunk 3]
```diff
@@ -58,12 +58,12 @@ void Season::addEpisode(Episode episode)
  * @returns matching Episode
  * @throws NotFound exception if no match found
  */
-Episode Season::getEpisode(QString name)
+Episode* Season::getEpisode(QString name)
 {
-    foreach (Episode episode, episodes) {
-        if (episode.name == name) {
-            return episode;
+    for(int i = 0; i < episodes.size(); i++) {
```
Comment (allanhaywood) [inline PlotTalk/Src/season.cpp:63]: Another convenient for loop is:
for (auto &value : values)
{
 ...
}

Which I believe, doesn't have the copy issue foreach does.

**Review by allanhaywood — COMMENTED** [review]
Lets sit down on Monday and go over the pointer/reference thing in class.

**Pull request merged**